components/eventsrv/src/main.rs

```diff
@@ -12,39 +12,122 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+extern crate env_logger;
 extern crate habitat_eventsrv;
 #[macro_use]
 extern crate log;
-extern crate env_logger;
-extern crate zmq;
 extern crate protobuf;
+extern crate zmq;
 
 mod message;
 
-use zmq::{Context, PULL, PUSH};
-use protobuf::parse_from_bytes;
-
 use message::event::EventEnvelope;
+use protobuf::parse_from_bytes;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::str::from_utf8;
+use zmq::{Context, SUB, XPUB};
 
 fn main() {
     let ctx = Context::new();
-    let pull_sock = ctx.socket(PULL).unwrap();
-    let push_sock = ctx.socket(PUSH).unwrap();
-    assert!(pull_sock.connect("tcp://127.0.0.1:34567").is_ok());
-    assert!(push_sock.connect("tcp://127.0.0.1:45678").is_ok());
+
+    // TODO: need to better parameterize these ports;
```

Comment:
There will be one EventSrv running per compute resource. Another way of saying this is - one EventSrv per supervisor on `n` instances/nodes/compute resources